### PR TITLE
feat(auth): Add ability to link a federated id with the UpdateUser() method.

### PR DESCRIPTION
**Pull request by rsgowman**

### CODE REVIEW

**Comment by rsgowman** [comment]
Note the diffbase. I'll merge+retarget to dev once the bulk_get stuff lands.

File: auth/user_mgt_test.go
Comment (hiranya911) [inline auth/user_mgt_test.go]: Also test for email and phone

File: integration/auth/user_mgt_test.go
Comment (hiranya911) [inline integration/auth/user_mgt_test.go]: Typically assertion failures are reported as follows.

```
t.Errorf("UpdateUser().ProviderUserInfo[*].UID = %v; want incldue = %q", providerUIDs, googleFederatedUID)
```

File: integration/auth/user_mgt_test.go
Comment (hiranya911) [inline integration/auth/user_mgt_test.go]: Perhaps move this part into a separate `t.Run()` context?

File: integration/auth/user_mgt_test.go
Comment (hiranya911) [inline integration/auth/user_mgt_test.go]: Shouldn't this fail with a validation error. We got:

```
        if len(u.params) == 0 {
		// update without any parameters is never allowed
		return nil, fmt.Errorf("update parameters must not be nil or empty")
	}
```

Perhaps we should change `ProvidersToDelete()` function ignore empty array, if the existing value for that key is already empty?

File: integration/auth/user_mgt_test.go
Comment (hiranya911) [inline integration/auth/user_mgt_test.go]: We really ought to stop passing `t` to helper functions. Instead return an error from this function, and handle it at the callsite.

File: integration/auth/user_mgt_test.go
Comment (hiranya911) [inline integration/auth/user_mgt_test.go]: `if auth.IsUserNotFound(err)`

File: integration/auth/user_mgt_test.go
Comment (rsgowman) [reply integration/auth/user_mgt_test.go:1201]: This pattern comes direct from go/go-style/best-practices#test-helper-error-handling.

I like it, since it reduces clutter at the call site for what's essentially a failed precondition. But I can change it to return an error if you feel strongly.

File: integration/auth/user_mgt_test.go
Comment (rsgowman) [reply integration/auth/user_mgt_test.go]: Done.

File: .github/scripts/publish_post_check.sh
Comment (hiranya911) [inline .github/scripts/publish_post_check.sh:1]: Please remove any changes to Actions workflow configs.

File: integration/auth/user_mgt_test.go
Comment (hiranya911) [reply integration/auth/user_mgt_test.go:1201]: Oh, that's good to know. I guess I'm always a bit weirded out by code that passes `t` around. But I suppose that's considered normal in Go.

File: integration/auth/user_mgt_test.go
Comment (hiranya911) [inline integration/auth/user_mgt_test.go:1211]: Error check can be inlined here.

**Review by hiranya911 — CHANGES_REQUESTED** [review]
Implementation looks pretty solid. Lets get rid of the CI/workflow stuff that's unrelated.

File: .github/scripts/publish_post_check.sh
Comment (rsgowman) [reply .github/scripts/publish_post_check.sh]: Done. (Must've been a bad merge)

File: integration/auth/user_mgt_test.go
Comment (rsgowman) [reply integration/auth/user_mgt_test.go]: Done.

**Review by hiranya911 — APPROVED** [review]
LGTM 👍

File: auth/user_mgt.go
Comment (egilmorez) [inline auth/user_mgt.go:335]: Assuming that none of this comment gets parsed into the generated docs.

Comment (rsgowman) [reply auth/user_mgt.go:335]: No, this shouldn't make it into the public facing docs.

**Pull request merged**
